### PR TITLE
Declare `self` as a reference

### PR DESCRIPTION
**Pull request by iamazy**

### CODE REVIEW

**Comment by iamazy** [comment]
@aleksuss PTAL,  sometimes `let txn = db.transaction();` and `txn.commit().unwrap()` maybe not in the same code block and add a `&` can make it possible

**Comment by aleksuss** [comment]
@iamazy, but at the same time `&` adds additional possibilities rather than only usage in different code blocks. For example, use a transaction more than once and usage in different threads. So I think you need to write tests for these use cases.

File: src/transactions/transaction.rs [hunk 1]
```diff
@@ -138,7 +138,7 @@ impl<'db, DB> Transaction<'db, DB> {
     /// [`Busy`]: crate::ErrorKind::Busy
     /// [`TryAgain`]: crate::ErrorKind::TryAgain
     /// [`Options::set_max_write_buffer_size_to_maintain`]: crate::Options::set_max_write_buffer_size_to_maintain
-    pub fn commit(self) -> Result<(), Error> {
+    pub fn commit(&self) -> Result<(), Error> {
```
[the remark this answers is not in the document]
Comment (iamazy) [reply src/transactions/transaction.rs:141]: Yes, but errors are difficult to handle currently. For example, if we use a transaction, the code should be

```rust
let txn = db.transaction();
match txn.put("key1", "value1") {
    Ok(v) => ...,
    Err(e) => txn.rollback(),
}
match txn.delete("key1") {
    Ok(v) => ...,
    Err(e) => txn.rollback(),
}
txn.commit().unwrap();
```

but if function `commit` receiving `&self`, the code could be

```rust

fn do_something(txn: &mut Txn) -> Result<(), Error> {
      txn.put("key1", "value1")?;
      txn.delete("key1")?;
      Ok(());
}

fn main() {
     let mut txn = db.transaction();
     match do_something(&mut txn) {
        Ok(()) => txn.commit().wrap(),
        Err(e) => txn.rollback(),
     }
}
```

I think the second way is more clear

[the remark this answers is not in the document]
Comment (aleksuss) [reply src/transactions/transaction.rs:141]: What's wrong with the "second way" now? I've just checked it out and it works fine.

[the remark this answers is not in the document]
Comment (iamazy) [reply src/transactions/transaction.rs:141]: oops, I used a wrong way, Thanks for your patient reply!!!

**Pull request closed without merging**